src/OpenLoco/GameCommands/LoadSaveQuit.cpp

```diff
@@ -0,0 +1,323 @@
+        Ui::WindowManager::close(Ui::WindowType::companyFaceSelection);
+        Ui::WindowManager::close(Ui::WindowType::objectSelection);
+        Ui::WindowManager::close(Ui::WindowType::saveGamePrompt);
+
+        clearScreenFlag(ScreenFlags::editor);
+        Audio::pauseSound();
+        Audio::unpauseSound();
+
+        if (Input::hasFlag(Input::input_flags::flag5))
+        {
+            Input::sub_407231();
+            Input::resetFlag(Input::input_flags::flag5);
+        }
+
+        Title::start();
+
+        throw GameException::Interrupt;
+    }
+
+    // 0x0043C427
+    void confirmSaveGame()
+    {
+        Input::toolCancel();
+
+        if (isEditorMode() && saveLandscapeOpen())
+        {
+            // 0x0043C4B3
+            auto path = fs::path(&_savePath[0]).replace_extension(S5::extensionSC5).u8string();
+            std::strncpy(&_currentScenarioFilename[0], path.c_str(), std::size(_currentScenarioFilename));
+
+            if (!S5::save(path, S5::SaveFlags::scenario))
+                Ui::Windows::Error::open(StringIds::landscape_save_failed, StringIds::null);
+            else
+                GameCommands::do_21(2, 0);
+        }
+        else if (!isNetworked() && saveSaveGameOpen())
+        {
+            // 0x0043C446
+            auto path = fs::path(&_savePath[0]).replace_extension(S5::extensionSV5).u8string();
+            std::strncpy(&_currentScenarioFilename[0], path.c_str(), std::size(_currentScenarioFilename));
+
+            S5::SaveFlags flags = {};
+            if (Config::get().flags & Config::flags::export_objects_with_saves)
+                flags = S5::SaveFlags::packCustomObjects;
+
+            if (!S5::save(path, flags))
+                Ui::Windows::Error::open(StringIds::error_game_save_failed, StringIds::null);
+            else
+                GameCommands::do_21(2, 0);
+        }
+        else
+        {
+            // 0x0043C511
+            GameCommands::do_72();
+            MultiPlayer::setFlag(MultiPlayer::flags::flag_2);
+
+            switch (_savePromptType)
+            {
+                case 0:
+                    MultiPlayer::setFlag(MultiPlayer::flags::flag_13); // intend to load?
+                    break;
+                case 1:
+                    MultiPlayer::setFlag(MultiPlayer::flags::flag_14); // intend to return to title?
+                    break;
+                case 2:
+                    MultiPlayer::setFlag(MultiPlayer::flags::flag_15); // intend to quit game?
+                    break;
+            }
+        }
+
+        // 0x0043C411
+        Gfx::invalidateScreen();
+    }
+
+    // 0x0043BFCB
+    static uint32_t loadSaveQuit(const uint8_t flags, uint8_t dl, uint8_t di)
+    {
+        if ((flags & GameCommandFlag::apply) == 0)
+            return 0;
+
+        if (dl == 1)
+        {
+            Ui::WindowManager::close(Ui::WindowType::saveGamePrompt);
+            return 0;
+        }
+
+        if (dl == 2)
+            return 0;
```

Comment:
In original this would go to 0x0043BFE3 why the change?

Reply:
I think I made a mistake here. I've changed the logic to match vanilla.